staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -211,8 +211,21 @@ func (le *LeaderElector) renew(ctx context.Context) {
 		timeoutCtx, timeoutCancel := context.WithTimeout(ctx, le.config.RenewDeadline)
 		defer timeoutCancel()
 		err := wait.PollImmediateUntil(le.config.RetryPeriod, func() (bool, error) {
-			return le.tryAcquireOrRenew(), nil
+			var succeed bool
+			done := make(chan struct{})
+			go func() {
+				defer close(done)
+				succeed = le.tryAcquireOrRenew()
+			}()
+
+			select {
+			case <-timeoutCtx.Done():
+				return false, fmt.Errorf("tryAcquireOrRenew timed out after %s", le.config.RenewDeadline)
```

Comment:
Hmm, now that this is based on the context, add `timeoutCtx.Err()` to the message.
The parent ctx passed to `renew` could've been cancelled.

Reply:
ok